packages/codemirror-blocks/spec/activation-test.ts

```diff
@@ -204,22 +197,28 @@ describe("deleting a node", () => {
   });
 });
 
+describe("Editing a node", () => {
+  it("should restore focus to the node after its edited", () => {
+    cmb.setValue("firstLiteral secondLiteral thirdLiteral");
+    userEvent.click(screen.getByRole("treeitem", { name: /secondLiteral/ }));
```

Comment:
When I first wrote this test, I used the `click()` function from simulate.ts and the test passed erroneously because `click()` uses `fireEvent.click()` which just sends a click event to the dom node.

A click event switches the node into edit mode, but does not call `activateByNid()`. So if there is just a click() event, then the focused node in the redux store will be out of date. Because the focused node was out of date, when the edited node was saved, the nodes useEffect() was triggered because the focused node (which is in the useEffect's dependency array) changed after the edit.

But in the browser, clicking on a node fires a bunch of events in sequence. mouseDown, then mouseUp, then click, etc, which each get handled in sequence. So in the browser, the state was being properly updated. The bug was that the `useEffect()` in the Node was skipped because nothing in the dependency array had changed.

To reproduce the browser's behavior in the test, I had to switch to using `userEvent.click()`, which does a better job of simulating all the actual events that get fired in a regular user interaction.